openeo_processes_dask/process_implementations/arrays.py

```diff
@@ -182,3 +182,75 @@ def array_labels(data: ArrayLike) -> ArrayLike:
     if len(data.shape) > 1:
         raise TooManyDimensions("array_labels is only implemented for 1D arrays.")
     return np.arange(len(data))
+
+
+def order(
+    data: ArrayLike,
+    asc: Optional[bool] = True,
+    nodata: Optional[bool] = None,
+    axis: Optional[int] = None,
+):
+    if isinstance(data, list):
+        data = np.asarray(data)
+    if len(data) == 0:
+        return np.nan
+    if axis is None:
+        axis = 0
+    if asc:
+        permutation_idxs = np.argsort(data, kind="mergesort", axis=axis)
+    else:  # [::-1] not possible
+        permutation_idxs = np.argsort(
+            -data, kind="mergesort", axis=axis
```

Comment:
This only works with numerical types, but will fail if trying to sort string or datetime arrays!
Why not just use `np.flip` on the found indices?

Reply:
I had used flip before, but switched to this approach because of the following side effect: 
Using the example in the documentation, order(data = [6,-1,2,null,7,4,null,8,3,9,9], asc = false, nodata = false) should result in [3, 6, 9, 10, 7, 4, 0, 5, 8, 2, 1].
With np.flip our result becomes [6, 3, 10, 9, 7, 4, 0, 5, 8, 2, 1], so the values that are the same are in the opposite order. Switching these after flip was used does not seem too convenient to me, what do you think?

Reply:
Ah, thanks for this - hmm, what we could do would be to count the number of nans before, only flip the array[index_last_nan:] and stich together again? Then we should have strings/datetimes covered too!

Reply:
I think what you are referring to would result in an array like [3, 6, 10, 9, 7, 4, 0, 5, 8, 2, 1], so the nan values would be handled. If there are values in the data twice, (like the "9" in the example), the order of these is also flipped. (The result in the example should be [3, 6, 10, 9, 10, 7, 4, 0, 5, 8, 2, 1]

Reply:
Hmm. @m-mohr what do you think about relaxing the `Ties will be left in their original ordering` requirement in the process spec for `order`? Is this strictly necessary for some usecase? Seems like implementation on our end would be simpler if this didn't have to be guaranteed.

Reply:
To get comparable and reproducible results, you can't let this undefined.

Reply:
Okay, that's fair, so we cannot drop this requirement completely - what about changing it such that ties will be in the original ordering if `asc=True` and flipped otherwise?
Otherwise I currently fail to see a way to do this such that arrays of strings or datetimes can also be ordered, we'd have to throw an error in these cases.

Reply:
@LukeWeidenwalker That's a good point, I think we can clarify that as I think that was the original intention anyway. Please open an issue in openeo-processes and I'll prepare a PR for 2.0.

Reply:
Opened https://github.com/Open-EO/openeo-processes/issues/409